File: tests/fixtures/sample.go
package main

import "fmt"

type Reader interface {
	Read(p []byte) (n int, err error)
}

type Closer interface {
	Close() error
}

type ReadCloser interface {
	Reader
	Closer
}

type UserService struct {
	Name string
}

func (s *UserService) Greet(user string) {
	fmt.Println(user)
}

func NewUserService(name string) *UserService {
	svc := &UserService{Name: name}
	svc.Greet(name)
	return svc
}

File: tests/fixtures/sample.rs
use std::fmt;

macro_rules! make_service {
    ($name:expr) => {
        UserService { name: String::from($name) }
    };
}

struct UserService {
    name: String,
}

impl UserService {
    fn greet(&self) {
        println!("{}", self.name);
    }
}

fn new_service(name: &str) -> UserService {
    let make = || make_service!(name);
    let svc = make();
    svc.greet();
    svc
}
